### PR TITLE
Issue #1342: setup-labels theme label catalog config-driven via .wholework.yml

### LINKED ISSUE #1342: setup-labels: theme label カタログを .wholework.yml 駆動でプロジェクト固有化
## Background

`/triage` (Step 6a, `skills/triage/SKILL.md:144-161`) は `gh label list` で `theme/*` ラベルを動的に取得し、各ラベルの `description` を分類基準として AI が Issue を振り分ける。この分類ロジック自体はプロジェクト非依存で汎用的に動く。

しかし実際に `theme/*` ラベルを作成しているのは `scripts/setup-labels.sh` の Always-group にハードコードされた wholework 自身のドッグフーディング用テーマ (`theme/observability`, `theme/ac-quality`, `theme/concurrency`, `theme/verify-backlog`, `theme/batch-orchestration`) の5つのみで、これは wholework 自身の自己開発バックログを分類するための固有ラベルである。他のプロジェクトが wholework を導入した場合、`setup-labels.sh` を実行すると自プロジェクトのドメインとは無関係なこれら5テーマが作成されてしまい、かつ自プロジェクト固有のテーマ (例: `theme/checkout`, `theme/payments`) を定義する手段が存在しない。

## Purpose

`.wholework.yml` でプロジェクト固有の theme label カタログを宣言できるようにし、`setup-labels.sh` がそこから `theme/*` ラベルを作成するようにする。`/triage` 側の分類ロジックは `gh label list` による動的取得のままで無改修。

## Proposal (Outline)

- `.wholework.yml` に `themes:` キー (name + description のリスト) を追加する。YAML の具体的な入れ子形式・bash でのパース方式は `/spec` で決定する
- `setup-labels.sh` は `.wholework.yml` に `themes:` が定義されている場合のみ、そこから `theme/{name}` ラベルを作成する。**`themes:` が未定義の場合は theme label を一切作成しない** (wholework 自身の5テーマを含め、既定で theme label を作成する挙動は廃止する)
- wholework 自身の既存5テーマ (`theme/observability` 等) は、本リポジトリ自身の `.wholework.yml` に `themes:` として明示的に移行する — 特別扱いはせず、他プロジェクトと同じ経路 (`themes:` 設定) を使う
- 既存ラベルの上書き挙動は `setup-labels.sh` の既存 `create_label()` 関数 (`gh label create ... --force`) をそのまま再利用する — 新しい sync/update ロジックは不要

## Design Decisions (/spec 非対話モードで自動解決)

Proposal が `/spec` に委ねた未決事項を、以下のとおり決定した (根拠は Spec の Notes を参照)。

- **YAML 形状**: リストではなく **block mapping** (`themes:` 配下に `{theme-name}: {description}` を1行1テーマ) を採用する。`.wholework.yml` の既存 `capabilities:` / `auto-retry-on-fail:` / `recoveries-auto-fire:` と同じ block mapping 形状で、`scripts/get-config-value.sh` の section-scan ループをそのまま流用できるため
- **色**: `theme/*` の色は `006B75` を `setup-labels.sh` 側の定数として固定する (config で指定させない)。namespace 単位の色一貫性を保つため
- **設定ファイルの解決**: `scripts/get-config-value.sh` と同じく CWD 基準の `.wholework.yml`。`WHOLEWORK_CONFIG_PATH` 環境変数による上書きに対応する (テストの決定性確保に必須)

### Pre-merge (auto-verified)

- [x]   `.wholework.yml` の `themes:` キーの使い方が `docs/guide/customization.md` に文書化されている
- [x]   `setup-labels.sh` が `.wholework.yml` の `themes:` からプロジェクト固有の `theme/{name}` ラベルを作成する
- [x]  `themes:` 未定義時に theme label が作成されないことが実装・テストで確認できる
- [x]    `themes:` 定義あり/なし双方のケースをカバーする新規テスト2件が `tests/setup-labels.bats` に追加され、全テストが pass する
- [x]   wholework 自身の既存5テーマ (`theme/observability`, `theme/ac-quality`, `theme/concurrency`, `theme/verify-backlog`, `theme/batch-orchestration`) が本リポジトリの `.wholework.yml` に `themes:` として明示移行されている
- [x]   `scripts/setup-labels.sh` のハードコードされた theme エントリが削除されている
- [x]    theme label の定義元がドキュメント側 (`docs/tech.md`, `docs/product.md` と各日本語ミラー, `modules/label-conventions.md`, `skills/triage/SKILL.md`) にも反映されている

### Post-merge

なし

### PR DESCRIPTION
**Pull request by saito**
## Summary

`scripts/setup-labels.sh` にハードコードされていた wholework 自身のドッグフーディング用 `theme/*` ラベル5件を、`.wholework.yml` の `themes:` キー駆動に置き換えました。

- `.wholework.yml` に `themes:` block mapping を追加し、既存5テーマ (`theme/observability` 等) を明示移行
- `scripts/setup-labels.sh`: `ALWAYS_LABELS` から `theme/*` 5件を削除し、`themes:` を読み込んで `theme/{name}` ラベルを description 付きで作成するパーサ + 作成ループを追加 (色は `006B75` 固定)。`themes:` 未定義時は theme label を一切作成しない
- `tests/setup-labels.bats`: `WHOLEWORK_CONFIG_PATH=/dev/null` を `setup()` に追加し、`themes:` 定義あり/なしの新規テスト2件を追加
- ドキュメント同期: `docs/guide/customization.md` / `docs/ja/guide/customization.md`、`docs/tech.md` / `docs/ja/tech.md` (Label Groups 表 Always 22→17)、`docs/product.md` / `docs/ja/product.md`、`modules/label-conventions.md`、`skills/triage/SKILL.md`

`/triage` Step 6a の分類ロジック (`gh label list` による動的取得) は無改修です。

## Verification (pre-merge)

- `.wholework.yml` の `themes:` キーの使い方が `docs/guide/customization.md` に文書化されている
- `setup-labels.sh` が `.wholework.yml` の `themes:` からプロジェクト固有の `theme/{name}` ラベルを作成する
- `themes:` 未定義時に theme label が作成されないことを実装・テストで確認
- `tests/setup-labels.bats` に新規テスト2件を追加、全22件 pass (`bats tests/setup-labels.bats`)
- wholework 自身の既存5テーマが `.wholework.yml` に `themes:` として明示移行されている
- `scripts/setup-labels.sh` のハードコードされた theme エントリが削除されている
- theme label の定義元がドキュメント側にも反映されている
- 全1715件の bats フルスイートが並列実行で pass (behavioral change detection により `.wholework.yml` / `docs/tech.md` を参照する他テストへの影響を確認)

## Verification (post-merge)

なし

closes #1342

### CODE REVIEW

File: scripts/setup-labels.sh [hunk 1]
```diff
@@ -10,7 +10,10 @@
 #   --no-fallback   Skip environment detection; only create always-group labels
```
Comment (saito) [inline scripts/setup-labels.sh]: [Documentation Consistency] The `--no-fallback` help text still reads "Skip environment detection; only create always-group labels". That was accurate while `theme/*` lived in `ALWAYS_LABELS`, but the new theme-group loop (lines 185-190) sits outside the `if [ "$NO_FALLBACK" = false ]` guard, so `--no-fallback` now also creates theme-group labels. This is user-visible via `--help` (prints lines 5-19), so the help text now contradicts the actual behavior.

Recommended fix: update line 10 to something like `#   --no-fallback   Skip environment detection; create always-group and theme-group labels only`.

File: docs/guide/customization.md [hunk 2]
```diff
@@ -153,6 +160,7 @@ This table is the **single source of truth (SSoT)** for all `.wholework.yml` con
...
 | `auto-stop-at` | string | `"verify"` | Declare the phase after which `/auto` should stop. Valid values: `spec`, `code`, `review`, `merge`, `verify`. Default `verify` runs the full pipeline. Use `review` for website projects where merge = publish and human gate before deploy is required. Per-invocation override: `--stop-at=<phase>`. Orthogonal to `autonomy:` tier. |
+| `themes` | block mapping | `{}` | Project-specific `theme/*` label catalog (`{name}: {description}`) consumed by `scripts/setup-labels.sh`. Unset means no `theme/*` labels are created at all — there is no default/fallback catalog (wholework's own dogfooding themes live in this repository's own `.wholework.yml`, not as a built-in default). Color is fixed at `006B75` for all theme labels; not configurable per theme. `/triage` Step 6a's classification logic reads the resulting `theme/*` labels dynamically via `gh label list` and is unaffected by this key directly. |
```
Comment (saito) [inline docs/guide/customization.md]: [Documentation Consistency] The new `themes` row documents shape, default, and fixed color, but omits two parse limitations the Spec explicitly accepted as known limitations: (1) everything from ` #` onward is stripped from a theme description, even inside a quoted scalar (so `checkout: "Theme: checkout # flow"` silently loses text); (2) only child lines matching `^[[:space:]]+[A-Za-z0-9._-]+[[:space:]]*:` are accepted — non-conforming theme names are silently skipped with no warning. Adjacent rows in this same SSoT table (e.g. `verify-ignore-paths`) already document this kind of limitation as a matter of convention.

Recommended fix: append the two limitations to the `themes` row (and its ja mirror at docs/ja/guide/customization.md:150).

File: tests/setup-labels.bats [hunk 3]
```diff
@@ -303,6 +308,41 @@ MOCK
     [ "$status" -ne 0 ]
 }
 
+# --- themes: (project-dependent theme label catalog) ---
```
Comment (saito) [inline tests/setup-labels.bats:311]: [Spec Deviation / Coverage] The two new tests cover "themes with 2 entries" and "themes key absent", but not two boundary cases the documentation now advertises as legal shapes: an empty `themes:` block mapping (`{}`), and `--no-fallback` combined with a populated `themes:` (the theme loop intentionally sits outside the `NO_FALLBACK` guard, but no test pins this contract). Not required by the Issue's ACs — coverage headroom, not a deviation.

Recommended fix: add a case for empty `themes:` asserting `count_label_creates == count_always_labels`, and extend/add a `--no-fallback` case with a populated config asserting `count_always_labels + N`.

File: scripts/setup-labels.sh [hunk 4]
```diff
@@ -93,6 +91,42 @@ FALLBACK_LABELS=(
...
+                    continue
+                    ;;
+                [[:space:]]*)
+                    if echo "$line" | grep -qE "^[[:space:]]+[A-Za-z0-9._-]+[[:space:]]*:"; then
+                        name=$(echo "$line" | sed -E "s/^[[:space:]]+([A-Za-z0-9._-]+)[[:space:]]*:.*$/\1/")
+                        desc=$(echo "$line" | sed -E "s/^[[:space:]]+[A-Za-z0-9._-]+[[:space:]]*:[[:space:]]*//" | sed -E "s/[[:space:]]+#.*$//" | sed -E "s/[[:space:]]*$//" | sed -E "s/^['\"]|['\"]$//" | sed -E "s/^['\"]|['\"]$//")
+                        THEME_LABELS+=("theme/${name}|${THEME_COLOR}|${desc}")
+                    fi
+                    ;;
+                *)
```
Comment (saito) [inline scripts/setup-labels.sh:125]: [Logic Error] A column-0 (non-indented) comment line inside the `themes:` block silently terminates parsing and drops every subsequent theme entry. The `case` statement has no branch for comment lines, so a `# comment` at column 0 falls into `*)` -> `IN_THEMES=false`. This diverges from the sibling parser `scripts/get-config-value.sh` (which skips `\#*` lines before section-boundary logic) and from the documented rule in `modules/detect-config-markers.md` ("Comment lines are ignored"). Empirically reproduced: given `themes:\n  a: "Theme A"\n# top-level comment\n  b: "Theme B"`, only `theme/a` is created; `theme/b` is silently dropped. This repo's own `.wholework.yml` uses column-0 comments between top-level keys as a matter of style, so this is a realistic authoring pattern for future theme entries.

Recommended fix: skip comment lines at the top of the loop body, mirroring `get-config-value.sh`: add `case "$line" in \\#*) continue ;; esac` before the `IN_THEMES` check.

File: scripts/setup-labels.sh [hunk 5]
```diff
@@ -93,6 +91,42 @@ FALLBACK_LABELS=(
     "value/5|BFD4F2|Value: 5"
 )
 
+# Theme-group labels: parsed from the `themes:` block mapping in .wholework.yml
```
Comment (saito) [inline scripts/setup-labels.sh:94]: [Logic Error / Cross-parser interaction] `themes:` is the first `.wholework.yml` block whose child key names are arbitrary user-chosen strings (existing sections like `capabilities:` have a fixed, known child-key set). `scripts/get-config-value.sh` resolves flat keys via `^[[:space:]]*${KEY}[[:space:]]*:`, which allows leading whitespace, so an indented theme child key can be matched as if it were a top-level config key. Empirically reproduced: with a synthetic `.wholework.yml` where `themes:` precedes a real key's own definition (or that key is otherwise unset), `bash scripts/get-config-value.sh <theme-name-matching-a-real-key> "DEFAULT"` returns the theme's description instead of the intended default. No live break in this repo today (`themes:` is last in the file, real keys are scanned first), but a downstream project naming a theme after an actual config key (e.g. `autonomy`, `permission-mode`, `always-pr`) that it leaves unset would get the theme description back instead of the documented default — silently.

Recommended fix: document in `docs/guide/customization.md`'s `themes` row that theme names must not collide with `.wholework.yml` config key names, since tightening `get-config-value.sh`'s flat-key anchor is a separate, wider-scoped change.

File: scripts/setup-labels.sh [hunk 6]
```diff
@@ -93,6 +91,42 @@ FALLBACK_LABELS=(
...
+if [ -f "$CONFIG_FILE" ]; then
+    IN_THEMES=false
+    while IFS= read -r line || [ -n "$line" ]; do
+        if [ "$IN_THEMES" = "true" ]; then
+            case "$line" in
+                "")
+                    continue
+                    ;;
+                [[:space:]]*)
+                    if echo "$line" | grep -qE "^[[:space:]]+[A-Za-z0-9._-]+[[:space:]]*:"; then
```
Comment (saito) [inline scripts/setup-labels.sh:117]: [Logic Error / UX] Lines inside `themes:` that don't match `^[[:space:]]+[A-Za-z0-9._-]+[[:space:]]*:` are silently skipped with no warning: a quoted key (`"checkout": "..."`), a key with characters outside `[A-Za-z0-9._-]` (e.g. `ui/ux: "..."`), or a nested sub-mapping (which instead produces incorrect extra labels, e.g. `theme/description`) all fail silently. The documented/supported shape (`name: "description"`) never trips this, so this is polish rather than a functional defect against the current spec.

Recommended fix: emit a `Warning: skipping unparsable themes entry: <line>` to stderr on the non-matching branch, and document the supported name character set in the `themes:` block's own comment (lines 94-99).

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| `.wholework.yml` の `themes:` キーの使い方が `docs/guide/customization.md` に文書化されている | PASS | `docs/guide/customization.md:109-114` に記載、Available Keys 表にも `themes` 行あり |
| `setup-labels.sh` が `.wholework.yml` の `themes:` から `theme/{name}` ラベルを作成する | PASS | `scripts/setup-labels.sh:94-127,186-190` |
| `themes:` 未定義時に theme label が作成されないことが実装・テストで確認できる | PASS | `THEME_LABELS` 空配列初期化 + `tests/setup-labels.bats` 新規テスト2件で確認 |
| `themes:` 定義あり/なし双方をカバーする新規テスト2件が追加され全テストが pass する | PASS | CI `Run bats tests` job SUCCESS (22/22 pass、レビューエージェントによるローカル再実行でも確認) |
| wholework 自身の既存5テーマが `.wholework.yml` に `themes:` として明示移行されている | PASS | `.wholework.yml:30-35` |
| `scripts/setup-labels.sh` のハードコードされた theme エントリが削除されている | PASS | `theme/observability` 等の記述なし |
| theme label の定義元がドキュメント側にも反映されている | PASS | `docs/tech.md` / `docs/ja/tech.md` / `docs/product.md` / `docs/ja/product.md` / `modules/label-conventions.md` / `skills/triage/SKILL.md` すべて更新済み |

## CI Status

| Job name | Status | Notes |
|---------|--------|-------|
| DCO | SUCCESS | — |
| Run bats tests (×2) | SUCCESS | — |
| Validate skill syntax (×2) | SUCCESS | — |
| Forbidden Expressions check (×2) | SUCCESS | — |
| Language Convention check (×2) | SUCCESS | — |
| macOS shell compatibility (×2) | SUCCESS | — |

全11ジョブ SUCCESS。

## Code Review

Base branch (`main`) との競合プリチェック: `origin/main` は本 PR のマージベースと一致しており、競合コンテキストなし。

review-spec (Opus) + review-bug×2 (diff scan / security scan) による並列レビューを実施。review-bug の指摘4件 (うち3件が review-spec と別観点) は Opus による2段階検証 (adversarial refutation) を実施し、1件を false positive として除外した。

### Verification Rejected (false positive filtering results)

- Rejected: `scripts/setup-labels.sh:186` theme creation loop positioned before Fallback-group under `set -euo pipefail` could abort before fallback labels are created → Reason: pre-existing behavior — the 5 `theme/*` entries lived inside `ALWAYS_LABELS` before this PR, in the exact same position ahead of the Fallback-group loop. The PR changes only the data source, not the ordering or failure semantics; not a new defect introduced by this PR.

### Issue Count Summary

- Posted as line comments: 7 (MUST: 0, SHOULD: 5, CONSIDER: 2)
- General Comments (in body): 0

---
Next: run `/merge 1344`.

### General Comments (auto-added: line outside PR diff range)
- **modules/detect-config-markers.md:70**: [Documentation Consistency] `docs/guide/customization.md` describes itself as the SSoT for all `.wholework.yml` keys and points readers here ("For the full reference including implementation details and YAML parsing rules, see `modules/detect-config-markers.md`") for the complete list. The new `themes` key was added to `customization.md`'s Available Keys table but has no counterpart row here, even though this file's own Purpose states "adding a new configuration item requires only adding a row to the marker definition table in this file", and other script-only keys (`verify-ignore-paths`, `patch-lock-timeout`) are already listed here as precedent.

Recommended fix: add a row after `auto-stop-at` (line 70), e.g. `| \`themes\` | (none — read directly by \`scripts/setup-labels.sh\`) | Block mapping of \`{name}: {description}\` | \`{}\` (no \`theme/*\` labels created) |`, or explicitly document in customization.md that non-variable keys are excluded from this table.

**Comment by saito** [comment]
Review: `docs/guide/customization.md` describes itself as the SSoT for all `.wholework.yml` keys and points readers to `modules/detect-config-markers.md` ("For the full reference including implementation details and YAML parsing rules") for the complete list. The new `themes` key was added to `customization.md`'s Available Keys table but has no counterpart row in `modules/detect-config-markers.md`, even though that file's own Purpose states "adding a new configuration item requires only adding a row to the marker definition table in this file", and other script-only keys (`verify-ignore-paths`, `patch-lock-timeout`) are already listed there as precedent. (path: modules/detect-config-markers.md, line: 70, severity: SHOULD — recorded as a General Comment because line 70 falls outside this PR's diff range.)

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| Logic Error | scripts/setup-labels.sh | 116 | Column-0 comment inside `themes:` block silently drops subsequent entries | Resolved | Skip comment lines at top of parser loop, mirroring `get-config-value.sh` |
| Logic Error / UX | scripts/setup-labels.sh | 110 | Unparsable `themes:` entries silently skipped with no warning | Resolved | Emit `Warning: skipping unparsable themes entry: ...` to stderr |
| Documentation Consistency | scripts/setup-labels.sh | 10 | `--no-fallback` help text says "only always-group labels" but theme-group is also created | Resolved | Updated help text |
| Documentation Consistency | docs/guide/customization.md | 163 | `themes` row omits documented parse limitations (`#` truncation, invalid name chars) | Resolved | Added limitations to `themes` row + ja mirror |
| Logic Error / Cross-parser interaction | scripts/setup-labels.sh | 94 | Theme names can collide with `.wholework.yml` config key names via `get-config-value.sh`'s flat-key lookup | Resolved | Documented the collision constraint in `customization.md` |
| Documentation Consistency | modules/detect-config-markers.md | 70 | `themes` key missing from the marker table `customization.md` points to as "the full reference" | Resolved | Added `themes` row + note |
| Spec Deviation / Coverage | tests/setup-labels.bats | 311 | New tests don't cover empty `themes:` or `--no-fallback` + populated `themes:` | Resolved | Added 3 new test cases (empty block, `--no-fallback`, column-0 comment regression) |

### Verification Rejected (false positive filtering results)

- Rejected: `scripts/setup-labels.sh:186` theme creation loop positioned before Fallback-group under `set -euo pipefail` could abort before fallback labels are created → Reason: pre-existing behavior — the 5 `theme/*` entries lived inside `ALWAYS_LABELS` before this PR, in the exact same position ahead of the Fallback-group loop. The PR changes only the data source, not the ordering or failure semantics.

### Lightweight Re-check Results

- `bats tests/setup-labels.bats`: PASS (25/25, including 3 new regression/edge-case tests)
- `python3 scripts/validate-skill-syntax.py skills/`: PASS (0 errors, 0 warnings)
- `bash scripts/check-forbidden-expressions.sh`: PASS (exit 0)
- `bash scripts/check-translation-sync.sh`: `docs/guide/customization.md` IN_SYNC (the 2 pre-existing gaps reported — `docs/guide/index.md` OUTDATED, `docs/guide/autonomy.md` MISSING_JA — are unrelated to this PR)
- New issues: none
